models/Coupons_model.php

```diff
@@ -182,28 +181,33 @@ public function minimumOrderTotal()
         return $this->min_total ?: 0;
     }
 
-    public function isExpired()
+    /**
+     * Check if a coupone is expired
+     *
+     * @param Carbon\Carbon $orderDateTime orderDateTime
+     *
+     * @return bool
+     */
+    public function isExpired($orderDateTime)
```

Comment:
For backward compatibility, make the `$orderDateTime` optional with a default value of null so we can use still use the current time when nothing is passed.

Something like this 

```
public function isExpired($orderDateTime = null) {
   if (is_null($orderDateTime))
      $orderDateTime = Carbon::now();
   ....
}
```